server/chat-plugins/github.ts

```diff
@@ -0,0 +1,266 @@
+/**
+ * Chat plugin to view GitHub events in a chatroom.
+ * By Mia, with design / html from xfix's original bot, https://github.com/xfix/GitHub-Bot-Legacy/
+ * @author mia-pi-git
+ */
+
+import {FS, Utils} from '../../lib';
+
+const STAFF_REPOS = ['pokemon-showdown', 'pokemon-showdown-client', 'Pokemon-Showdown-Dex'];
+const COOLDOWN = 10 * 60 * 1000;
+
+export const gitData: GitData = JSON.parse(FS("config/chat-plugins/github.json").readIfExistsSync() || "{}");
+
+interface GitHookHandler {
+	on(event: 'pull_request', callback: (repo: string, ref: string | undefined, result: PullRequest) => void): void;
+	on(event: 'push', callback: (repo: string, ref: string, result: Push) => void): void;
+	on(event: string, callback: (repo: string, ref: string | undefined, result: any) => void): void;
+	listen(): void;
+	server: import('http').Server;
+}
+
+interface Push {
+	commits: Commit[];
+	sender: {login: string};
+	compare: string;
+}
+
+interface PullRequest {
+	action: string;
+	number: number;
+	pull_request: {
+		url: string,
+		html_url: string,
+		title: string,
+		user: {
+			login: string,
+			html_url: string,
+		},
+		merge_commit_sha: string,
+	};
+	sender: {login: string};
+}
+
+interface Commit {
+	id: string;
+	message: string;
+	author: {name: string, avatar_url: string};
+	url: string;
+}
+
+interface GitData {
+	usernames?: {[username: string]: string};
+	bans?: {[username: string]: string};
+}
+
+export const GitHub = new class {
+	readonly hook: GitHookHandler | null = null;
+	updates: {[k: string]: number} = Object.create(null);
+	constructor() {
+		try {
+			this.hook = (require('githubhook'))(Config.github);
+		} catch {}
+		this.listen();
+	}
+	listen() {
+		if (!this.hook) return;
+		this.hook.listen();
+		this.hook.on('push', (repo, ref, result) => this.handlePush(repo, ref, result));
+		this.hook.on('pull_request', (repo, ref, result) => this.handlePull(repo, ref, result));
+	}
+	private getRepoName(repo: string) {
+		switch (repo) {
+		case 'pokemon-showdown':
+			return 'server';
+		case 'pokemon-showdown-client':
+			return 'client';
+		case 'Pokemon-Showdown-Dex':
+			return 'dex';
+		default:
+			return repo.toLowerCase();
+		}
+	}
+	handlePush(repo: string, ref: string, result: Push) {
+		const branch = /[^/]+$/.exec(ref)?.[0] || "";
+		if (branch !== 'master') return;
+		const messages = {
+			staff: [] as string[],
+			development: [] as string[],
+		};
+		for (const commit of result.commits) {
+			const {message, url} = commit;
+			let shortened = /.+/.exec(message)?.[0] || "";
+			if (message !== shortened) {
+				shortened += '…';
+			}
+			const username = this.getUsername(commit.author.name);
+			const repoName = this.getRepoName(repo);
+			const id = commit.id.substring(0, 6);
+			messages.development.push(
+				Utils.html`[<font color='FF00FF'>${repoName}</font>] <a href=\"${url}\"><font color='606060'>${id}</font></a> ${shortened} <font color='909090'>(${username})</font>`
```

Comment:
Although PS isn't generally HTML5 anyway, we should be using <span> tags with the style tag applied rather than the deprecated <font>. Something like ``<span style="color:#hexvalue">text</span>`` should work fine.

Reply:
Fixed.